src/peergos/server/mutable/UserBasedBlacklist.java

```diff
@@ -0,0 +1,128 @@
+package peergos.server.mutable;
+
+import peergos.shared.cbor.*;
+import peergos.shared.corenode.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.merklebtree.*;
+import peergos.shared.mutable.*;
+import peergos.shared.storage.*;
+import peergos.shared.user.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.*;
+
+public class UserBasedBlacklist implements PublicKeyBlackList {
+
+    private static final long RELOAD_PERIOD = 3_600_000;
+
+    private Map<PublicKeyHash, Boolean> banned = new ConcurrentHashMap<>();
+    private final CoreNode core;
+    private final MutablePointers mutable;
+    private final ContentAddressedStorage dht;
+    private final Path source;
+    private final ForkJoinPool pool = new ForkJoinPool(1);
+    private long lastModified, lastReloaded;
+
+    public UserBasedBlacklist(Path source, CoreNode core, MutablePointers mutable, ContentAddressedStorage dht) {
+        this.source = source;
+        this.core = core;
+        this.mutable = mutable;
+        this.dht = dht;
+        pool.submit(() -> {
```

Comment:
I'd suggest  making this more resilient  to failures:
```
while (true) {
                try {
                    updateBlackList();
                } catch (Throwable t) {
                    //log t
                }

                try {
                    Thread.sleep(5000);
                } catch (InterruptedException e) {}
            }
```

Reply:
Also I think there's a race-condition here: if ``this.isAllowed`` is called before the pool executes ``updateBlacklist`` for the first time. We could add a synchronous  call of ``updateBlacklist`` to the constructor before  submitting the runnable to ``pool``?

Reply:
Yeah this is not thread safe. E.g. I clear the map before updating it, without a lock. But I'm not stressed about exact guarantees, so long as it works 99% of the time.